File: BUEN_SABOR_FRONTEND/src/models/Promocion.ts
import DetallePromocion from "./DetallePromocion";
import type  TipoPromocion from "./enums/TipoPromocion";
import type ImagenPromocion from "./ImagenPromocion";
import type Sucursal from "./Sucursal";
export default class Promocion {
    id?: number;
    denominacion: string = "";
    fechaDesde: Date = new Date();
    fechaHasta: Date = new Date();
    horaDesde: string = "";
    horaHasta: string = "";
    descripcionDescuento: string = "";
    precioPromocional: number = 0;
    tipoPromocion!: TipoPromocion;
    imagenes: ImagenPromocion[] = [];
    detalles: DetallePromocion[] = [];
    activa: boolean = false;
    sucursales!: Sucursal[];
    eliminado!: boolean;
}

File: BUEN_SABOR_BACKEND/src/main/java/com/lab4/buen_sabor_backend/model/Promocion.java
package com.lab4.buen_sabor_backend.model;

import com.fasterxml.jackson.annotation.*;
import com.lab4.buen_sabor_backend.model.enums.TipoPromocion;
import jakarta.persistence.*;
import lombok.*;

import java.time.*;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Promocion extends Master {

    private String denominacion;
    private LocalDate fechaDesde;
    private LocalDate fechaHasta;
    private LocalTime horaDesde;
    private LocalTime horaHasta;
    private String descripcionDescuento;
    private Double precioPromocional;

    @Enumerated(EnumType.STRING)
    private TipoPromocion tipoPromocion;

    private Boolean activa = false;

    @ManyToMany
    @JoinTable(
            name = "promocion_sucursal",
            joinColumns = @JoinColumn(name = "promocion_id"),
            inverseJoinColumns = @JoinColumn(name = "sucursal_id")
    )
    @JsonIgnore
    private Set<Sucursal> sucursales = new HashSet<>();

    @OneToMany(mappedBy = "promocion", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<ImagenPromocion> imagenes = new HashSet<>();

    @OneToMany(mappedBy = "promocion", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<DetallePromocion> detalles = new ArrayList<>();
}

